drools-shapes/drools-shapes-generator-plugin/src/main/java/org/drools/shapes/OntoModelCompiler.java

```diff
@@ -359,14 +359,20 @@ public boolean streamMetaclasses( boolean withDefaultClasses ) {
         MetaclassModel metaModel = (MetaclassModel) mcompiler.compile( model, withDefaultClasses );
 
         boolean success = metaModel.save( getJavaDir().getPath() );
+        String regexSeparator = File.separator;
+        if (regexSeparator.equals("\\"))
```

Comment:
Strictly speaking, the error can happen anywhere the replaceAll method is used, and either of the regex strings contains a control character  (i.e. the escape character '\'). It happens that in this case it is the result of an OS dependent value. As to being more robust, I think calling out the specific character  (as opposed to looking at the OS value) is the more robust answer.